avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/Array_of_BOOLEAN_GenericDeserializer_5988037297456147388_5988037297456147388.java

```diff
@@ -18,36 +18,32 @@ public Array_of_BOOLEAN_GenericDeserializer_5988037297456147388_5988037297456147
         this.readerSchema = readerSchema;
     }
 
-    public List<Boolean> deserialize(List<Boolean> reuse, Decoder decoder)
+    public PrimitiveBooleanList deserialize(PrimitiveBooleanList reuse, Decoder decoder)
         throws IOException
     {
-        List<Boolean> array738 = null;
-        long chunkLen739 = (decoder.readArrayStart());
-        if (chunkLen739 > 0) {
-            List<Boolean> arrayReuse740 = null;
-            if ((reuse) instanceof List) {
-                arrayReuse740 = ((List)(reuse));
+        PrimitiveBooleanList array734 = null;
+        long chunkLen735 = (decoder.readArrayStart());
+        if (chunkLen735 > 0) {
+            PrimitiveBooleanList arrayReuse736 = null;
+            if ((reuse) instanceof ColdPrimitiveBooleanList) {
+                arrayReuse736 = ((ColdPrimitiveBooleanList)(reuse));
             }
-            if (arrayReuse740 != (null)) {
-                arrayReuse740 .clear();
-                array738 = arrayReuse740;
+            if (arrayReuse736 != (null)) {
+                arrayReuse736 .clear();
+                array734 = arrayReuse736;
             } else {
-                array738 = new org.apache.avro.generic.GenericData.Array<Boolean>(((int) chunkLen739), readerSchema);
+                array734 = new ColdPrimitiveBooleanList(((int) chunkLen735));
             }
             do {
-                for (int counter741 = 0; (counter741 <chunkLen739); counter741 ++) {
-                    Object arrayArrayElementReuseVar742 = null;
-                    if ((reuse) instanceof GenericArray) {
-                        arrayArrayElementReuseVar742 = ((GenericArray)(reuse)).peek();
```

Comment:
This object allocation caused by the peeking is removed in the new version of the generated code.

Reply:
how far do you want to go? if the underlying is a real primitive list Systems.arraycopy() is even faster :-)

Reply:
I don't think `arraycopy()` applies here... the element peeking happens only so that the element from the old list can be recycled (mutated) with a new value.